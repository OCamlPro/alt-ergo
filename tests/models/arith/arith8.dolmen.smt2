; This test checks if the optimization doesn't stop after getting an optimized
; model in a fixed branch of the SAT solver.
(set-option :produce-models true)
(set-option :verify-models false)
(set-logic ALL)
(declare-const x Int)
(declare-const y Int)
; If the SAT solver explores first the right branch of the following OR gate, it
; shouldn't conclude that x = 2 is an optimized model.
(assert (or (<= x 4) (<= x 2)))
(maximize x)
; The current implementation of SatML explores the right branch of OR gates
; first. We add the following line to be sure the optimization feature still
; works after changing the order of branches.
(assert (or (<= y 2) (<= y 4)))
(maximize y)
(check-sat)
(get-model)
(get-objectives)
