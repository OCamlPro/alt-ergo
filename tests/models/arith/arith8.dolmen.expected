
unknown
(
  (define-fun x () Int 4)
  (define-fun y () Int 4)
)
(objectives 
  (x 4)
  (y 4)
)